float32[3] angular_velocity
float32[3] specific_force